src/SecTester.Bus/Internal/JsonHttpMethodEnumerationStringConverter.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SecTester.Bus.Internal;
+
+public class JsonHttpMethodEnumerationStringConverter : JsonConverter<HttpMethod>
+{
+  private static readonly HttpMethod _patch = new("PATCH");
+  private static readonly HttpMethod _copy = new("COPY");
+  private static readonly HttpMethod _link = new("LINK");
+  private static readonly HttpMethod _unlink = new("UNLINK");
+  private static readonly HttpMethod _purge = new("PURGE");
+  private static readonly HttpMethod _lock = new("LOCK");
+  private static readonly HttpMethod _unlock = new("UNLOCK");
+  private static readonly HttpMethod _propfind = new("PROPFIND");
+  private static readonly HttpMethod _view = new("VIEW");
+  private static readonly HttpMethod _trace = new("TRACE");
```

Comment:
That map implicitly does verifiction of the incoming value for its  validity like an `Enum` type. Also it prevents deserializer from redundant object creation. I believe since the `HttpMethod` internally implemented relying on static memebers we may use the same approach.

Reply:
Also, suggest to verify the `Method` coming from the client side and fail serialization to save roundtrip to server.

Reply:
If so, why don't use reflection to avoid mistyping or any further changes:

```csharp
  private static readonly IEnumerable<HttpMethod> BaseMethods = typeof(HttpMethod)
    .GetProperties(BindingFlags.Public |
                   BindingFlags.Static |
                   BindingFlags.DeclaredOnly)
    .Select(f => f.GetValue(null))
    .Cast<HttpMethod>();

  private static readonly IEnumerable<HttpMethod> CustomMethods = new List<HttpMethod>
  {
    new("PATCH"),
    new("COPY"),
    new("LINK"),
    new("UNLINK"),
    new("PURGE"),
    new("LOCK"),
    new("UNLOCK"),
    new("PROPFIND"),
    new("VIEW")
  };
```

Then you can concat them and convert a result to a dictionary

```csharp
CustomMethods.Concat(BaseMethods).ToDictionary(x => x.ToString(), x => x);
```

However, you need to resolve the issue with duplicates (e.g. there is Patch in NET 6):

```csharp
private static IDictionary<string, HttpMethod> GetMethods()
{
  if (Methods is not { Count: 0 })
  {
    return Methods;
  }

  var methods = CustomMethods.Concat(BaseMethods);
    
  foreach (var method in methods)
  {
    Methods[method.Method] = method;
  }

  return Methods;
}
```
So, if they add new methods to HttpMethod, we will use them (i.e. `HttpMethod.Link.Should().BeSameAs(expected)`)